src/trapfunc.cpp

```diff
@@ -222,56 +219,74 @@ bool trapfunc::beartrap( const tripoint &p, Creature *c, item * )
     return true;
 }
 
-bool trapfunc::board( const tripoint &, Creature *c, item * )
+bool trapfunc::board( const tripoint &p, Creature *c, item * )
 {
     if( c == nullptr ) {
         return false;
     }
-    // tiny animals don't trigger spiked boards, they can squeeze between the nails
-    if( c->get_size() == creature_size::tiny ) {
+    map &here = get_map();
+    // Only a chance to step on the board, dependent on size
+    if( !x_in_y( occupied_tile_fraction( c->get_size() ), 1.0f ) ) {
         return false;
     }
     c->add_msg_player_or_npc( m_bad, _( "You step on a spiked board!" ),
                               _( "<npcname> steps on a spiked board!" ) );
     monster *z = dynamic_cast<monster *>( c );
-    Character *you = dynamic_cast<Character *>( c );
     if( z != nullptr ) {
         if( z->has_effect( effect_ridden ) ) {
             add_msg( m_warning, _( "Your %s stepped on a spiked board!" ), c->get_name() );
             get_player_character().mod_moves( -z->get_speed() * 0.8 );
         } else {
             z->mod_moves( -z->get_speed() * 0.8 );
         }
-        z->deal_damage( nullptr, bodypart_id( "foot_l" ), damage_instance( damage_cut, rng( 3,
-                        5 ) ) );
-        z->deal_damage( nullptr, bodypart_id( "foot_r" ), damage_instance( damage_cut, rng( 3,
-                        5 ) ) );
+        z->deal_damage( nullptr,
+                        random_entry( c->get_ground_contact_bodyparts() ),
+                        damage_instance( damage_cut, rng( 3, 5 ) ) );
     } else {
-        int total_cut_dmg = 0;
-
-        for( const bodypart_id &bp : c->get_ground_contact_bodyparts() ) {
-            dealt_damage_instance dd = c->deal_damage( nullptr, bp, damage_instance( damage_cut, rng( 6,
-                                       10 ) ) );
-            total_cut_dmg += dd.type_damage( damage_cut );
-        }
-        if( !you->has_flag( json_flag_INFECTION_IMMUNE ) && total_cut_dmg > 0 ) {
+        dealt_damage_instance dd = c->deal_damage( nullptr,
+                                   random_entry( c->get_ground_contact_bodyparts() ),
+                                   damage_instance( damage_cut, rng( 3, 5 ) ) );
+        int total_cut_dmg = dd.type_damage( damage_cut );
+        Character *you = c->as_character();
+        if( you != nullptr && !you->has_flag( json_flag_INFECTION_IMMUNE ) && total_cut_dmg > 0 ) {
             const int chance_in = you->has_trait( trait_INFRESIST ) ? 256 : 35;
             if( one_in( chance_in ) ) {
                 you->add_effect( effect_tetanus, 1_turns, true );
             }
         }
     }
     c->check_dead_state();
+    // Weight of 100kg+ is guaranteed to break the trap, linear chance as weight increases
+    if( x_in_y( c->get_weight() / 1_kilogram, 100 ) ) {
+        // destroy trap
+        here.remove_trap( p );
+        if( !c->is_avatar() ) {
+            // stupid branch to properly capitalize the start of the sentence
```

Comment:
Why don't you use `disp_name` for both cases? `disp_name( false, true )` for monsters will give you `The <monster> destroys...`. `disp_name( false, true )` for NPCs will give you `John Doe destroys...`

Reply:
Simple answer: I found out afterwards and hadn't bothered to change it

![image](https://github.com/user-attachments/assets/025cea3a-3695-4ca9-bbf7-6788d5b0abd1)